adaptivefiltering/segmentation.py

```diff
@@ -130,124 +174,189 @@ def __init__(self, dataset=None, segmentation=None):
             )
-        # prepare the map data
 
-        # if a dataset is given the boundary of the data set is calculated via the hexbin function
-        # and the in memory dataset is converted into EPSG:4326.
-        if dataset and segmentation is None:
-            self.segmentation = self.get_boundary(dataset)
-
-        elif dataset is None and segmentation:
-            self.segmentation = segmentation
+        # check if dataset and segmentation are of correct type
+        if dataset:
+            if isinstance(dataset, Segmentation):
+                raise Exception(
+                    "A segmentation was given as a dataset, please call Map(segmentation=yourSegmentation)"
+                )
+            elif not isinstance(dataset, DataSet):
+                raise Exception(
+                    f"The given dataset is not of type DataSet, but {type(dataset)}."
+                )
+
+        elif segmentation:
+            if isinstance(segmentation, DataSet):
+                raise Exception(
+                    "A DataSet was given as a Segmentation, please call Map(dataset=yourDataset)"
+                )
+            elif not isinstance(segmentation, Segmentation):
+                raise Exception(
+                    f"The given segmentation is not of type Segmentation, but {type(segmentation)}."
+                )
+
+        # convert to pdal dataset
+        if dataset:
+
+            dataset = PDALInMemoryDataSet.convert(dataset)
+            # preserve the original srs from dataset
+            if in_srs is None:
+                self.original_srs = dataset.spatial_reference
+            else:
+                if in_srs is None:
+                    raise Exception(
+                        "No srs could be found. Please specify one or use a dataset that includes one."
+                    )
+                self.original_srs = in_srs
+
+        self.dataset = dataset  # needed for overlay function.
+
+        # convert to a srs the ipyleaflet map can use.
+        # the only way this seems to work is to convert the dataset to EPSG:4326 and set the map to expect EPSG:3857
+        # https://gis.stackexchange.com/questions/48949/epsg-3857-or-4326-for-googlemaps-openstreetmap-and-leaflet/48952#48952
+        boundary_segmentation = self.load_hexbin_boundary(dataset, segmentation)
+
+        # get square edges of the boundary_segmentation for use in hillshade overlay
+        self.boundary_edges = get_min_max_values(boundary_segmentation)
+
+        self.setup_map(boundary_segmentation)
+        self.setup_controls()
+
+        # set up overlay dict.
+        # this stores the parameters used in the load_overlay function to avoid multipole calculations of the same overlay
+
+        self.overlay_dict = {}
+
+    def load_overlay(
+        self,
+        map_type,
+        classification=asprs[:],
+        resolution=2,
+        azimuth=315,
+        angle_altitude=45,
+        opacity=0.6,
```

Comment:
This parameter list is a sort of redundant repetition of the input parameters for visualization. Instead, it would be possible to accept `**kwargs` and validate it against the schema defined in `schema/visualization.json`. The docstring could just refer to it as "accepts all parameters that `DataSet.show` accepts. The non-supported visualization types (Scatter, 2.5D mesh) can be excluded by throwing an error.